### ISSUE TITLE
Button configure in project doesn't work

### ISSUE THREAD

**Issue by jet-snowman** [Strider-CD/strider#861]

**Comment by jet-snowman** [comment]
just need to type / in view

**Comment by knownasilya** [comment]
Can you explain a little more?

**Comment by jet-snowman** [comment]
sure, current URL is http://domain/account/project/config - it doesn't work, but if append / like  http://domain/account/project/config/ works great 08.10.2015, 18:53, "Ilya Radchenko" <notifications@github.com>:Can you explain a little more?—Reply to this email directly or view it on GitHub.   -- Regards, Vadim Sadykhov

**Comment by knownasilya** [comment]
So which button is it that takes you to this url without the /?

**Comment by jet-snowman** [comment]
sorry, I forgot to send image, http://s27.postimg.org/udiy6i2xv/Screen_Shot_2015_10_08_at_11_57_06_AM.png

**Comment by knownasilya** [comment]
What version are you using?

**Comment by jet-snowman** [comment]
1.7.5 master branch and after updated from 1.6.6 to 1.7.5 I lost all settings projects and hooks don't work...

**Comment by knownasilya** [comment]
Did you update all of your plugins? There haven't been any changes to the schema so all of your settings should stay as is.

I suggest a fresh install with your current DB, just to rule out any old dependency issues.

**Comment by jet-snowman** [comment]
I did - git pull origin master, npm install and all was broken... it's very very very not good

**Comment by knownasilya** [comment]
How are you running the app?

**Comment by jet-snowman** [comment]
via forever, so I think problem also in browserify

when I run npm install
The package browserify does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer browserify-shim@3.8.2 wants browserify@>= 2.3.0 < 9
npm ERR! peerinvalid Peer minifyify@7.1.0 wants browserify@>= 5

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ec2-user/site/strider/npm-debug.log
[ec2-user@dev strider]$ browserify --version
11.2.0

**Comment by knownasilya** [comment]
Try deleting your whole strider folder and recloning and npm installing, then run with `npm start`

**Comment by jet-snowman** [comment]
did, npm install was good. So result:
1) I lost all settings
2) Web hooks still aren't working - maybe bitbucket has problems now
3) button still doesn't work :))

**Comment by jet-snowman** [comment]
by the way, notification works but for only owner(or one account) so if users have an access of the same project only one will get email, don't know why

**Comment by knownasilya** [comment]
What browser are you using?

**Comment by jet-snowman** [comment]
chrome and in opera, safari don't work, but in version 1.7.2 the button is working... I think if type / all will work

**Comment by knownasilya** [comment]
So my button doesn't have a `/` but it works right (it redirects to a url with a slash). Very weird.

**Comment by knownasilya** [comment]
I just submitted a commit to fix the /, 9031e41701193771e17e67a064e9dd8b59d24645

**Comment by knownasilya** [comment]
It working in 1.7.2 and not in 1.7.5 makes no sense whatsoever..

**Comment by jet-snowman** [comment]
yeah problem not with /, when I type / in url works because browser reloads your page, I made video  https://play.boomstream.com/page.html?code=X1Ld3Pvv

**Comment by knownasilya** [comment]
Thanks a bunch! FYI, we don't explicitly support Opera (no one does lol).

**Comment by jet-snowman** [comment]
problem not in Opera, the key works on grid as you can see, but when I login in project button doesn't work. I think problem in JS with routers. Also Opera almost is the same as safari, chrome, webkit engine )

**Comment by knownasilya** [comment]
I'll try a fresh install tomorrow and let you know how it fares.

**Comment by oliversalzburg** [comment]
1. [Your video link](https://play.boomstream.com/page.html?code=X1Ld3Pvv) leads to a 404 error page for me.
2. The **Configure** button behaves fine for me on Opera 32.0.1948.69 Windows 8.1 x64.
3. When the browser requests `https://ourserver.name/account/project/config`, Strider will respond with a 301 Moved Permanently with a location of `/account/project/config/` (it adds the trailing slash).

**Comment by oliversalzburg** [comment]
@knownasilya I get the same `EPEERINVALID` error when running `npm install`

**Comment by knownasilya** [comment]
> IMHO, relying on semver caret matching and having people clear their node_modules is not nice :(

Agreed, I've been doing that less and less across my projects. It's unfortunate that some projects don't follow semver, when the project they depend on does.

**Comment by oliversalzburg** [comment]
In our projects we only use fixed versions for the dependencies and `~` for devDependencies. However, that would not have made any difference here :D

The key point I wanted to get across is, always declare the required minimum version for the dependency, even if `~` or `^` should match it.

**Comment by knownasilya** [comment]
I just ran into the issue after restarting the browser.

**Comment by oliversalzburg** [comment]
I faintly remember having an issue with trailing slashes as well. But I don't see how it is actually possible, since we're using https://github.com/avinoamr/connect-slashes

**Comment by knownasilya** [comment]
I think it might be a bad client built..

**Comment by oliversalzburg** [comment]
Interesting. Now I get this as well. Clicking the **Configure** button does not create a request to the server.

**Comment by oliversalzburg** [comment]
Problem was introduced with 4e008b17ffc6b325f282024b2e9d66430368c331 Reverting the change fixes the issue.

**Comment by knownasilya** [comment]
You only reverted the change on line 70? Does the issue it was intending to fix still fixed?

**Comment by oliversalzburg** [comment]
@knownasilya When I click job IDs in the build list, I see the log for that build. So, I assume, yeah, the issue is still fixed. But I never experienced it in the first place.

**Comment by knownasilya** [comment]
The issue was that when you refresh the page it wouldn't go back to the selected job, I think.

**Comment by oliversalzburg** [comment]
Ah, I see. Yeah, that's broken.

**Comment by knownasilya** [comment]
Latest version of master should be fixed

**Comment by oliversalzburg** [comment]
Heh, now I got both bugs ;)

```js
TypeError: Cannot read property 'params' of undefined
    at job.js:75
    at Scope.$broadcast (angular.js:12952)
    at afterLocationChange (angular.js:9907)
    at angular.js:9895
    at Scope.$eval (angular.js:12673)
    at Scope.$digest (angular.js:12485)
    at Scope.$apply (angular.js:12777)
    at HTMLHtmlElement.<anonymous> (angular.js:9849)
    at HTMLHtmlElement.jQuery.event.dispatch (jquery.js:4435)
    at HTMLHtmlElement.elemData.handle (jquery.js:4121)
```

**Comment by knownasilya** [comment]
Ok, I fully reverted the PR.
